Upgrade to rails 6.1 and ruby 2.7

SuperTux88 opened the pull request:
Rails 5.2 and ruby 2.6 are both end-of-life. So this upgrades to supported versions again. I also upgraded to sprockets 4 and updated a few other gems that were not possible before (blocked by old rails/ruby version).

SuperTux88 committed:
Bump js_image_paths to support rails 6 and sprockets 4

SuperTux88 committed:
Bump diaspora_federation to support rails 6

SuperTux88 committed:
Upgrade to sprockets 4

SuperTux88 committed:
Move ActsAsTaggableOn::Tag overrides to initializer

This is to prepare for zeitwerk autoloader, and the old file couldn't be
autoloaded anyway, so the easiest is to just move it out of the models
folder.

SuperTux88 committed:
Don't load sidekiq workers during initialization

This is to prevent the warning about zeitwerk autoloader

SuperTux88 committed:
Upgrade to rails 6.0

SuperTux88 committed:
Fix deprecation warnings for rails 6.0

SuperTux88 committed:
Use template_name instead of action_name for notification mails

This is a new feature in rails 6

SuperTux88 committed:
Remove workaround for schema.rb

We don't have the schema.rb anymore in the repo, so we don't need it
compatible with both databases anymore.

See 5b1be7d8dac07eb133250835bce12e36f8e4908a

SuperTux88 committed:
Upgrade to rails 6.1

SuperTux88 committed:
Fix deprecation warnings for rails 6.1

SuperTux88 committed:
Remove "Did you mean?" from api errors when a parameter is missing

This isn't helpful at all for an api if you don't send a required
parameter and get an error response that just tells you what parameters
that were available.

This is a new feature with rails >= 6.1 and ruby >= 2.7, so this just
keeps the old behaviour of older rails/ruby versions.

SuperTux88 committed:
Add ruby 2.7, drop ruby 2.5

SuperTux88 committed:
Bump nokogiri

SuperTux88 committed:
Bump yajl-ruby

SuperTux88 committed:
Bump acts-as-taggable-on

SuperTux88 committed:
Bump sidekiq and sidekiq-cron

SuperTux88 committed:
Fix rendering of error pages to remove deprecation warning for rails 6.0

Flaburgan commented:
Thank you for this much needed work!
I'm surprised you're targeting a minor diaspora* version for a big change like Rails upgrade, you feel confident enough about it?

SuperTux88 commented:
We already did a ruby upgrade in an earlier minor update (0.7.14.0), and since ruby 2.6 is EOL, and rails 5.2 doesn't support ruby 2.7, we don't have much choice than doing it. But I kept the changes so it doesn't contain breaking changes (the rest of the rails stuff can be done on `develop` only for the next major). So it should be fine for a minor update and it allows podmins to install a ruby version again which still gets updates. So the only breaking change is dropping ruby 2.5 (which is EOL since even longer).

SuperTux88 committed:
Update yarn files for rails 6

Also allow `yarnpkg` binary for `bin/yarn`

SuperTux88 committed:
Upgrade to bundler 2

SuperTux88 commented:
I also updated bundler to bundler 2 now. This is maybe also a small breaking change, as podmins also need to update bundler ... but we have `gem install bundler` in our update-guide anyway (we just need to remove the `1.17.3` version number when releasing 0.7.18.0), and if podmins update to ruby 2.7 anyway (which is recommended), it's easier to install the current bundler instead of the old one, so I think that's fine.

denschub commented on .rubocop.yml:
```diff
@@ -57,6 +57,11 @@ Layout/SpaceAroundEqualsInParameterDefault:
 # are needed.
 Style/StringLiterals:
   EnforcedStyle: double_quotes
+  Exclude:
+    - "config/application.rb"
+    - "config/boot.rb"
+    - "config/environment.rb"
+    - "config/environments/*.rb"
```
I assume you excluded those files to allow for single-quotes in the config files to keep the diff to the Rails-original sources minimal? If so, a comment stating as much probably would be a good idea, because I was about to write a comment asking why you flipped the quotes in those files. :)

SuperTux88 replied:
Yes, that's exactly the reason, because when updating rails it also generates new versions of these files, and I wanted to keep them as close to the original as possible to see what actually changed ... but yes, I'll add a comment here :+1:

skyschub reviewed (approved):
This is kinda huge, and my review quality probably reflects that. But I didn't spot anything bad, so let's ship it!

SuperTux88 committed:
Fix follow up tasks for assets:precompile when no manifest existed

When no `.sprockets-manifest-xxx.json` existed, every instance of
`Sprockets::Manifest` generated their own path with their own random
filename, and since this happened before the assets were actually
precompiled, they were all empty. So the error pages didn't find the
manifest and the non-digest assets also didn't have any assets to copy.

So lets create our own instance of `Sprockets::Manifest` here, AFTER
`assets:precompile`, which then loads the manifest json that was used
during precompile, so all precompiled assets are available.

SuperTux88 commented:
Merged as many commits to `next-minor` and 4edaebb94f10bb3ab616398ff89c27820dfaef9d, e7e34a8c24ad6527660e308592117e0a6bd03176 and 03d2001cf2ee0bc93d5e413b30387382ec13e1e9 to `develop` only.

Thanks for the review @denschub :cookie:
